src/Illuminate/Foundation/PackageManifest.php

```diff
@@ -55,7 +55,7 @@ public function __construct(Filesystem $files, $basePath, $manifestPath)
         $this->files = $files;
         $this->basePath = $basePath;
         $this->manifestPath = $manifestPath;
-        $this->vendorPath = $basePath.'/vendor';
+        $this->vendorPath = getenv('COMPOSER_VENDOR_DIR') ?: $basePath.'/vendor';
```

Comment:
`getenv` is not a good function to use (because it's not threadsafe, so can leak variables across different websites running on the same server, or resolve to nothing, even after `putenv` is called). We should use `env` or just `$_SERVER`.

Reply:
Thanks! Made changes.